src/os/bluestore/BlueStore.cc

```diff
@@ -6000,21 +6000,17 @@ BlueStore::TransContext *BlueStore::_txc_create(OpSequencer *osr)
   return txc;
 }
 
-void BlueStore::_txc_update_store_statfs(TransContext *txc)
+void BlueStore::_txc_update_store_statfs(TransContext *txc, TransContext::volatile_statfs *vstatfs)
 {
   if (txc->statfs_delta.is_empty())
     return;
 
-  logger->inc(l_bluestore_allocated, txc->statfs_delta.allocated());
-  logger->inc(l_bluestore_stored, txc->statfs_delta.stored());
-  logger->inc(l_bluestore_compressed, txc->statfs_delta.compressed());
-  logger->inc(l_bluestore_compressed_allocated, txc->statfs_delta.compressed_allocated());
-  logger->inc(l_bluestore_compressed_original, txc->statfs_delta.compressed_original());
+  vstatfs->allocated() += txc->statfs_delta.allocated();
+  vstatfs->stored() += txc->statfs_delta.stored();
+  vstatfs->compressed() += txc->statfs_delta.compressed();
+  vstatfs->compressed_allocated() += txc->statfs_delta.compressed_allocated();
+  vstatfs->compressed_original() += txc->statfs_delta.compressed_original();
 
-  bufferlist bl;
-  txc->statfs_delta.encode(bl);
-
-  txc->t->merge(PREFIX_STAT, "bluestore_statfs", bl);
```

Comment:
I don't think this change is safe, either, because rocksdb could choose to persist one of these earlier transactions (with an actual change) before persisting the final sync transaction (with the batched stats changes), and then teh system crashes, and the stats are left out of sync.